OpenRA.Game/MiniYaml.cs

```diff
@@ -511,8 +519,15 @@ static IReadOnlyCollection<MiniYamlNode> MergeSelfPartial(IReadOnlyCollection<Mi
 
 		static MiniYaml MergePartial(MiniYaml existingNodes, MiniYaml overrideNodes)
 		{
-			existingNodes?.Nodes.ToDictionaryWithConflictLog(x => x.Key, "MiniYaml.Merge", null, x => $"{x.Key} (at {x.Location})");
-			overrideNodes?.Nodes.ToDictionaryWithConflictLog(x => x.Key, "MiniYaml.Merge", null, x => $"{x.Key} (at {x.Location})");
+			lock (ConflictScratch)
```

Comment:
The code below looks more complex. You are also introducing a lock here. Seems unelegant.

Is there enough benefit.

Does merging also happen in game? Or during mod loading at startup? Or mostly when using the utils. 

(Wouldn't it be an idea to let all yaml files use a shared dictionary where keys are replaced by symbols/the hash of keys. To allow for faster compare. It could also reduce network transfer sizes.)

Reply:
It occurs during loading - so initial game loading, plus loading when you are starting a match.

The lock is somewhat inelegant, but in practise we never load from multiple threads so it won't be a bottleneck.

<details>

<summary>Benchmark Script</summary>

```c#
[MemoryDiagnoser]
public class BenchmarkJob
{
	static readonly string[] Rules = new[]
	{
		"ra|rules/misc.yaml",
		"ra|rules/ai.yaml",
		"ra|rules/player.yaml",
		"ra|rules/palettes.yaml",
		"ra|rules/world.yaml",
		"ra|rules/defaults.yaml",
		"ra|rules/vehicles.yaml",
		"ra|rules/husks.yaml",
		"ra|rules/structures.yaml",
		"ra|rules/infantry.yaml",
		"ra|rules/civilian.yaml",
		"ra|rules/decoration.yaml",
		"ra|rules/aircraft.yaml",
		"ra|rules/ships.yaml",
		"ra|rules/fakes.yaml",
	};

	IEnumerable<IReadOnlyCollection<MiniYamlNode>> sources;

	[GlobalSetup]
	public void Setup()
	{
		var stringPool = new HashSet<string>(); // Reuse common strings in YAML
		sources = Rules
			.Select(s => s.Replace("ra|", @"<file-path-here>\OpenRA\mods\ra\"))
			.Select(s => MiniYaml.FromStream(File.OpenRead(s), s, stringPool: stringPool)).ToArray();
	}

	[Benchmark]
	public List<MiniYamlNode> MiniYamlMerge()
	{
		return MiniYaml.Merge(sources);
	}
}
```
</details>

Before
| Method        | Mean     | Error    | StdDev   | Gen0      | Gen1    | Allocated |
|-------------- |---------:|---------:|---------:|----------:|--------:|----------:|
| MiniYamlMerge | 30.93 ms | 0.328 ms | 0.307 ms | 6593.7500 | 31.2500 |  79.24 MB |

After
| Method        | Mean     | Error    | StdDev   | Gen0      | Gen1    | Allocated |
|-------------- |---------:|---------:|---------:|----------:|--------:|----------:|
| MiniYamlMerge | 23.41 ms | 0.223 ms | 0.208 ms | 2375.0000 | 31.2500 |  28.46 MB |

We already pool strings keys when parsing. Changing them to symbols would require working out what to do for the consumers of the result - who are currently able to use the strings and slice and dice them as they see fit.